app/Models/Department.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Department extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+      'office_id',
+      'name',
+      'parent_department_id',
```

Comment:
This relationship hasn't been added. It is a relationship _back_ to `Department` itself, and defines the "parent" department of this department. For example, a "Testing" Department that is a child of a "Research" department, which in turn is a child of the "Head" Department, if that makes sense.
```php
public function parent_department() {
    return $this->belongsTo(Department::class, "parent_department_id");
}
```